File: app/src/main/java/com/waygo/data/base/route/DatabaseRoute.java
package com.waygo.data.base.route;

import android.net.Uri;
import android.support.annotation.NonNull;

import rx.functions.Action1;

public interface DatabaseRoute {
    @NonNull String getPath();
    @NonNull String getTableName();
    @NonNull String getDefaultSortOrder();
    @NonNull String getWhere(Uri uri);
    @NonNull String getMimeType();
    void notifyChange(@NonNull Uri uri, @NonNull Action1<Uri> notifyChange);
}

File: app/src/main/java/com/waygo/data/base/route/DatabaseRoute.kt
package com.waygo.data.base.route

import android.net.Uri

import rx.functions.Action1

public interface DatabaseRoute {
    public fun getPath(): String
    public fun getTableName(): String
    public fun getDefaultSortOrder(): String
    public fun getWhere(uri: Uri): String
    public fun getMimeType(): String
    public fun notifyChange(uri: Uri, notifyChange: (Uri) -> Unit)
}
